docs/synapse/userguides/ug015_storm_ref_pivot.rst

```diff
@@ -11,85 +15,339 @@ Where specific query examples are given, they are commonly provided in pairs usi
+* If no parameters are specified, ``refs()`` will return the combined results of both ``refs(in)`` and ``refs(out)`` (e.g., execute all pivots to / from the working set of nodes).
+
+**Operator syntax:**
+
+.. parsed-literal::
+  **refs(** [ **in** | **out** ] **)**
+
+**Macro syntax:**
+
+N/A
+
+**Examples:**
+
+* Return all of the nodes that **reference** a set of nodes:
+  ::
+    refs( in )
+
+  Assume a set of ``inet:fqdn`` nodes in the working set. ``refs(in)`` will return any node with a secondary property type *<ptype> = <valu>* that matches the *<type> = <valu>* of those ``inet:fqdn`` nodes. For example, this may include ``inet:dns:a`` nodes (``inet:dns:a:fqdn``), ``inet:whois:rec`` nodes (``inet:whois:rec:fqdn``), additional ``inet:fqdn`` nodes (``inet:fqdn:domain``), etc.
+
+* Return all the nodes **referenced by** a set of nodes:
+  ::
+    refs( out )
+
+  Assume a set of ``inet:dns:a`` nodes in the working set. ``refs(out)`` will return any node with a primary *<type> = <valu>* that matches any secondary property type *<ptype> = <valu>* in the working set. As an ``inet:dns:a`` record includes secondary properties of type ``inet:fqdn`` (``inet:dns:a:fqdn``) and ``inet:ipv4`` (``inet:dns:a:ipv4``), the query may return those node types.
+
+* Return all of the nodes that **reference** or are **referenced by** a set of nodes:
+  ::
+    refs()
+
+  Assume a set of ``inet:email`` nodes in the working set. An ``inet:email`` *<type> = <valu>* may be referenced by *<ptype> = <valu>* from a variety of forms, including ``inet:whois:contact`` (``inet:whois:contact:email``), ``inet:dns:soa`` (``inet:dns:soa:email``) or ``inet:web:acct`` (``inet:web:acct:email``). Based on its secondary properties, a *<ptype> = <valu>* from an ``inet:email`` node may reference *<type> = <valu>* forms such as ``inet:fqdn`` (``inet:email:fqdn``) or ``inet:user`` (``inet:email:user``).
+
+**Usage notes:**
+
+* ``refs()`` / ``refs(in)`` / ``refs(out)`` can be useful in an "exploratory" manner to identify what other nodes / forms are "reachable from" (can be pivoted to or from) the working set of nodes. However, because ``refs()`` essentially carries out all possible pivots, the set of nodes returned may be quite large. In such cases a more focused ``pivot()`` or ``join()`` operation may be more useful.
 
 fromtags()
 ----------
-Todo
+Given a working set that contains one or more ``syn:tag`` nodes, returns the specified set of nodes to which those tags have been applied.
+
+``fromtags()`` can be thought of as pivoting **from** a set of **tags**, to a set of nodes that have those tags.
+
+Optional parameters:
+
+*  **<form>:** return only nodes of the specified form(s).
+* If no forms are specified, ``fromtags()`` returns all nodes for all forms to which the tags are applied.
+
+**Operator syntax:**
+
+.. parsed-literal::
+  
+  **fromtags(** [ *<form_1>* **,** *<form_2>* **,** *...<form_n>* ] **)**
```

Comment:
`limit` also works on fromtags but does *not* work with totags

Reply:
Fixed